echopop/live/live_survey.py

```diff
@@ -0,0 +1,343 @@
+        )
+        # ---- Add cloud storage options, if needed
+        self.config.update({"storage_options": cloud_storage_options})
+
+        # Initialize input attribute
+        self.input = copy.deepcopy(LIVE_DATA_STRUCTURE["input"])
+
+        # Initialize database attribute
+        self.database = copy.deepcopy(LIVE_DATA_STRUCTURE["database"])
+
+        # Initialize the results attribute
+        self.results = copy.deepcopy(LIVE_DATA_STRUCTURE["results"])
+
+        # Initialize the extrapolation grid
+        initialize_grid(self.config)
+
+        # Add database paths to configuration attribute
+        eldp.configure_database_paths(self.config)
+
+        # Configure the spatial settings
+        self.input.update({"spatial": eldl.configure_spatial_settings(self.config)})
+
+        # TODO: Add verbosity for printing database filepaths/connections
+        if verbose:
+            pass
+
+    def __repr__(self):
+
+        # Get any acoustic files created
+        if "acoustic_files" in self.meta["provenance"]:
+            # ---- Get the filenames
+            acoustic_filenames = self.meta["provenance"]["acoustic_files_read"]
+            # ---- Subset if many files are being processed
+            if len(acoustic_filenames) > 2:
+                acoustic_filenames = (
+                    acoustic_filenames[:2] + ["..."] + [f"[n = {len(acoustic_filenames)}]"]
+                )
+            # ---- Format string
+            acoustic_files = ", ".join(acoustic_filenames)
+        else:
+            acoustic_files = "None"
+
+        # Get any biology files created
+        if "biology_files" in self.meta["provenance"]:
+            # ---- Get the filenames
+            biology_filenames = self.meta["provenance"]["biology_files_read"]
+            # ---- Subset if many files are being processed
+            if len(biology_filenames) > 4:
+                biology_filenames = biology_filenames + ["..."]
+            # ---- Format string
+            biology_files = ", ".join(biology_filenames)
+        else:
+            biology_files = "None"
+
+        # Get linked database names
+        linked_dbs = "\n   ".join(
+            [f"{key.title()}: {db}" for key, db in self.config["database"].items()]
+        )
+
+        return (
+            f"LiveSurvey-class object \n"
+            f"Timestamp: {self.meta['date']} \n"
+            f"Acoustic files being processed: \n   {acoustic_files}\n"
+            f"Biology files being processed: \n   {biology_files}\n"
+            f"Linked databases: \n   {linked_dbs}"
+        )
+
+    def __str__(self):
+        return self.__repr__()
+
+    def load_acoustic_data(
+        self, xarray_kwargs: dict = {}, input_filenames: Optional[list] = None, verbose: bool = True
+    ):
+
+        # Validate the data directory and format the filepaths
+        acoustic_files = eldl.validate_data_directory(
+            self.config, dataset="acoustics", input_filenames=input_filenames
+        )
```

Comment:
Note to self: only new files are in the returned `acoustic_files`, because `validate_data_directory` right now contains `query_processed_files` that would exclude files that have already been processed.